encodings/fsst/src/array.rs

```diff
@@ -54,6 +57,14 @@ impl FSSTArray {
             vortex_bail!(InvalidArgument: "symbols and symbol_lengths arrays must have same length");
         }
 
+        if uncompressed_lengths.len() != codes.len() {
+            vortex_bail!(InvalidArgument: "uncompressed_lengths must be same len as codes");
+        }
+
+        if !uncompressed_lengths.dtype().is_int() {
```

Comment:
extremely minor, but might be worth asserting min >= 0 or that it's unsigned

Reply:
I'll add that in flup